### PR TITLE
Feat/add kubapi restrictions

### PR DESCRIPTION
**Pull request by garloff**
This implements #246.

### CODE REVIEW

**Comment by garloff** [comment]
Sidenote: The patches here are relative to a branch that I constructed to work against the latest, not-yet-merged state.
This makes it easier to review.
Technically, a merge to the branch does not require a review but would be useless.
So, please look at the implementation here.
When we are all fine, we would probably have to rebase against the then current state of main and require a formal review in order to really merge.

**Review by chess-knight — COMMENTED** [review]
I found that this is not supported by OVN provider. See https://cluster-api-openstack.sigs.k8s.io/clusteropenstack/configuration.html#restrict-access-to-the-api-server or https://docs.openstack.org/octavia/xena/user/feature-classification/index.html#operation_allowed_cidr. I suggest to mention this somewhere.

File: terraform/files/bin/apply_kubeapi_cidrs.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+{
+	if test -z "$1" -o "$1" = "null"; then return 0; fi
+	if test "$1" = "[]" -o "$1" = "[ ]" -o "$1" = "[  ]"; then return 0; fi
+	return 1
+}
+
+get_own_fip()
+{
+	NETS=$(openstack server list --name "${PREFIX}-mgmtcluster" -f value -c Networks)
+	FIP=${NETS##*, }
```
Comment (chess-knight) [inline terraform/files/bin/apply_kubeapi_cidrs.sh:19]: I found the same problem as [here](https://github.com/SovereignCloudStack/k8s-cluster-api-provider/issues/342) with `]}` at the end

Comment (garloff) [reply terraform/files/bin/apply_kubeapi_cidrs.sh:19]: Good point, need to make this work with openstackclient versions >= and < 5.5.

We don't offer ovn LB in front of kube-api yet, the cloud.conf settings only affect occm created LBs (k8s objects of service type LoadBalancer).
So we can mention it, but it's not something that anyone would run into without extra manual tuning.
Still: good catch!

Comment (chess-knight) [reply terraform/files/bin/apply_kubeapi_cidrs.sh:19]: > We don't offer ovn LB in front of kube-api yet, the cloud.conf settings only affect occm created LBs (k8s objects of service type LoadBalancer).

Thank you for pointing this out. CAPO has to be configured specially to use ovn lb-provider.

Comment (garloff) [reply terraform/files/bin/apply_kubeapi_cidrs.sh]: Comment has been added and (more importantly) the regexp adjusted to work with both pre-5.5 and 5.5+ openstackclien network output format.t

**Comment by garloff** [comment]
I will close this in favor of #368 which is a rebase against current main.

**Pull request closed without merging**